### PR TITLE
LOG-2009: Rename functional framework to remove "fluentd"

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
Rename functional framework to remove 'fluentd' from name

cc @vimalk78 @ajaygupta978

### Links
* https://issues.redhat.com/browse/LOG-2009

### CODE REVIEW

File: test/framework/functional/framework.go [hunk 1]
```diff
@@ -52,16 +52,16 @@ type FluentdFunctionalFramework struct {
 	collector CollectorFramework
 }
 
-func NewFluentdFunctionalFramework() *FluentdFunctionalFramework {
+func NewCollectorFunctionalFramework() *CollectorFunctionalFramework {
```
[the remark this answers is not in the document]
Comment (jcantrill) [reply test/framework/functional/framework.go:55]: There should be no reason to create a second framework for vector; we are functionally testing the same output via blackbox testing.  Ideally, with the difference of something simple like a flag, the framework should be able to test either without difference.  The only differences at this time which would be collector specific is:
* the receiver (we should be able to converge here)

[the remark this answers is not in the document]
Comment (jcantrill) [reply test/framework/functional/framework.go:55]: This change is only a rename of the overall framework

[the remark this answers is not in the document]
Comment (jcantrill) [reply test/framework/functional/framework.go:55]: @vimalk78 this commit builds off this change in the direction we want to go https://github.com/jcantrill/cluster-logging-operator/commit/26e56302b09e420e04431d3d4593c07b1530b08b  (currently broken, but it does deploy)

**Review by ajaygupta978 — COMMENTED** [review]
Looks fine

**Pull request merged**